core/src/test/scala/org/apache/spark/deploy/DecommissionWorkerSuite.scala

```diff
@@ -242,8 +242,10 @@ class DecommissionWorkerSuite
       assert(jobResult === 2)
     }
     // 6 tasks: 2 from first stage, 2 rerun again from first stage, 2nd stage attempt 1 and 2.
-    val tasksSeen = listener.getTasksFinished()
```

Comment:
Would you happen to recall the github actions error you got that lead to this change ? I would like to dig further because I invoke the listener using TestUtils.withListener(sc, listener): Which waits for the listener to drain and also removes the listener. 

So I don't think wrapping this in an eventually should actually be doing anything: The listener has already been removed. Perhaps I ought to bring back the the "waiting for job done" inside of the getTasksFinished or as a separate call. 

I would like to understand further just so that I can learn about some of the gotchas with this listener stuff.

Reply:
Yeah good point, it's probably not the listener. It's only showing up for me in GHA though - https://github.com/apache/spark/pull/29367/checks?check_run_id=972990200#step:14:13579

Reply:
Lets undo this change then. I am rerunning this PR locally to debug. Thanks for sharing the GHA link. It helps.

Reply:
I think I know the race. I will file another PR for this either against this PR or against master. The race is simply that we need to wait for the decommissioning to have happened before triggering the fetch failure. On a busy machine, the listener can be delayed.

Reply:
Sounds good, I'll back this change out.

Reply:
This will take me a while to fix. I will make this fix against the master branch. 

Apparently #29211 broke some of my state keeping that I was relying on in #29032 :-P. Let me think through how to fix this for real. So I think the test failure is real and it is worrisome that it isn't failing as frequently as it should. 

Stay tuned for a PR to fix this but in the meanwhile please back out this test change. Thanks for surfacing this issue.

Reply:
Okay so @HyukjinKwon also reported a test failure: https://github.com/apache/spark/pull/29014#issuecomment-672642497 and that is encouraging. I will work on a fix for this ASAP.